### PR TITLE
SageMaker pipeline

### PR DESCRIPTION
**Pull request by joeranbosma**
Dear @Yukiya-Umimi,

We have been training your models, but unfortunately ran into some issues. We would like to propose the following changes to your codebase:
- Bugfix mha2nnunet settings to resample to `3.0x0.5x0.5` and crop to `[24, 384, 384]`: https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/commit/443f703298ac3fc5a953d192ab57d8e51b30c858
- Make repository pip-installable for easier importing of functions: https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/commit/446ebc688fbcf0cbbf1c8ff13a89fb771273211c
- Restructure Docker container definition: https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/commit/a479e03d6fd118a1bc73122bcab68e428935fa8d
- Add training code for segmentation model: https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/commit/29d8416b8d3201a4064d652a9ca772bed8fa466b

Additionally, I would like to propose some minor changes:
- Cleaned up Python cache files
- Created `__init__.py` files for making the repo pip-installable
- Change imports, e.g. `from utils import ...` to `from segmentation.utils import ...`  for making the repo pip-installable
- Fix version of `scikit-image` to `0.20.0` to ensure `skimage.metrics` exists
- Remove unused imports

I've put all of these proposed changes in this PR. Could you go through the changes, and if your agree, merge them into `main`?

Thanks!

### CODE REVIEW

**Review by Yukiya-Umimi — APPROVED** [review]

**Comment by Yukiya-Umimi** [comment]
Hi,


It is OK,  I have great confidence in your work!


Best,
Honey K


| |
yukiya
|
|
***@***.***
|
---- Replied Message ----
| From | Joeran ***@***.***> |
| Date | 3/26/2023 20:22 |
| To | ***@***.***> |
| Cc | ***@***.***> ,
***@***.***> |
| Subject | [Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge] SageMaker pipeline (PR #10) |

Dear @Yukiya-Umimi,

We have been training your models, but unfortunately ran into some issues. We would like to propose the following changes to your codebase:

Bugfix mha2nnunet settings to resample to 3.0x0.5x0.5 and crop to [24, 384, 384]: 443f703
Make repository pip-installable for easier importing of functions: 446ebc6
Restructure Docker container definition: a479e03
Add training code for segmentation model: 29d8416

Additionally, I would like to propose some minor changes:

Cleaned up Python cache files
Created __init__.py files for making the repo pip-installable
Change imports, e.g. from utils import ... to from segmentation.utils import ... for making the repo pip-installable
Fix version of scikit-image to 0.20.0 to ensure skimage.metrics exists
Remove unused imports

I've put all of these proposed changes in this PR. Could you go through the changes, and if your agree, merge them into main?

Thanks!

You can view, comment on, or merge this pull request online at:

  https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/pull/10

Commit Summary
8b436b9 Cleanup
4449bac Bugfix
7503fda Safer with file permissions
1780970 Cleanup
29d8416 Supervised segmentation training script
bbede2f Cleanup imports
df290cb Cleanup imports
9c905a5 Docker container for segmentation training
65e3468 Update scikit-image
3f82db4 Cleanup imports
8ff71f1 Add missing dependency
4c37241 Cleanup imports
ed52843 Cleanup imports
1bc1ab3 Bugfix
443f703 Bugfix preprocessing resampling
a878c2b Move Docker container definitions
b88a08d Add preprocessing Docker container
4845190 Temporarily use this branch
cea67e5 Cleanup
a073b20 Cleanup & init files
446ebc6 Make repository pip-installable
a479e03 Cleanup Docker containers & bugfix
File Changes

(34 files)

R.dockerignore (0)
ADockerfile (18)
Abuild.sh (4)
Aclassification/__init__.py (0)
Dclassification/__pycache__/config.cpython-37.pyc (0)
Dclassification/__pycache__/trainer.cpython-37.pyc (0)
Dclassification/__pycache__/utils.cpython-37.pyc (0)
Apreprocess/__init__.py (0)
Mpreprocess/preprocess.py (9)
Apyproject.toml (22)
Arequirements.txt (16)
Asegmentation/__init__.py (0)
Dsegmentation/__pycache__/config.cpython-37.pyc (0)
Dsegmentation/__pycache__/data_loader.cpython-37.pyc (0)
Dsegmentation/__pycache__/loss.cpython-37.pyc (0)
Dsegmentation/__pycache__/metrics.cpython-37.pyc (0)
Dsegmentation/__pycache__/model.cpython-37.pyc (0)
Dsegmentation/__pycache__/trainer.cpython-37.pyc (0)
Dsegmentation/__pycache__/utils.cpython-37.pyc (0)
Msegmentation/config.py (5)
Msegmentation/data_loader.py (9)
Msegmentation/predict_2d.py (15)
Msegmentation/requirements.txt (3)
Msegmentation/run.py (11)
Msegmentation/trainer.py (33)
Msegmentation/utils.py (8)
Asetup.cfg (54)
Asetup.py (18)
Dsubmit_result/__pycache__/old_unet.cpython-37.pyc (0)
Msubmit_result/process.py (20)
Mtrain_1_classification.py (8)
Mtrain_2_segmentation.py (56)
Dtraining_dockers/1_classification/Dockerfile (23)
Dtraining_dockers/1_classification/build.sh (4)
Patch Links:
https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/pull/10.patch
https://github.com/Yukiya-Umimi/ITUNet-for-PICAI-2022-Challenge/pull/10.diff
